.github/workflows/build.yml

```diff
@@ -41,6 +41,12 @@ jobs:
           ${{ runner.os }}-go-
     - name: Run test
       run: make test
+    - name: Upload Test Coverage
```

Comment:
Sending data to Codecov is more than just "enabling code coverage" and needs explanation. To continue reviewing these pull requests I need you to include some accompanying description of intent and other considerations and not just code.

This step depends on a secret that's not defined. Presumably you expect us to go sign up add it before merging this?

Have you reviewed alternatives? Is there a decent GitHub Action that might handle this without the additional service dependency, or is a fully hosted coverage tool worth it?

Code spells out _how_ something will happen but not necessarily _what_ or _why_. A PR's description should always speak to the questions of _what_ and _why_ so the review can consider whether the code accomplishes those goals effectively.

Reply:
I've removed everything concerning code coverage from this PR to make it focused on the linters. 
Codecov is the most commonly used coverage tool on the project I've seen on github. There are probably other solutions but to my knowledge all of them require to send a file summarizing the result of the coverage of your files.

> This step depends on a secret that's not defined. Presumably you expect us to go sign up add it before merging this?

I took it from [here](https://github.com/codecov/codecov-action#usage), It seems unnecessary when your repository is public and it doesn't seems to fail when this token is missing but if it's not used it's better to not include it